### DISCUSSION TITLE
Start date / end date for contract agreement missing

### DISCUSSION THREAD

**Discussion by DrDanielMetz** [eclipse-tractusx/tractusx-edc#631] [Q&A] [answered]
I checked on the AgreementAPI and on the database of EDC 0.5.0:

 There is no startDate and endDate now for the contract agreement. Is this by intention?

Before, it was possible to see, when a contract starts and when it ends. With EDC 0.3.x there was even a possiblity to provide a "duration" in the contract definition, to define the duration of a contract.

Is this changed?
How long a contract is valid now?
How I can get this information now from the API?

 The background is that we wanted to check if a contract already exists and reuse it, instead of always re-negotiate new contracts.

**Comment by ndr-brt** [comment, marked as answer]
Expiration can be expressed through a policy constraint, there's a detailed documentation here:
https://eclipse-edc.github.io/docs/#/submodule/Connector/docs/developer/contract-duration/contract-validity-check

**Reply by DrDanielMetz** [reply]
Yes, I got this documentation on the Eclipse element chat from @paullatzelsperger around two weeks back.

**Comment by lgblaumeiser** [comment]
Closed as stale, please renew item if this is still an issue
